### ISSUE TITLE
Pulped corpses are rising

### ISSUE THREAD

**Issue by l29ah** [CleverRaven/Cataclysm-DDA#30772]
# Steps To Reproduce

Load the save and walk around. The pulped corpses near the Humvee will rise.

 - Game Version: 0.D-3344-gf3049e5eea-dirty

# Additional context
[2019.04.29-4.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3210296/2019.04.29-4.zip)

**Comment by Night-Pryanik** [comment]
Didn't test your save, but tested on latest experimental.
Can't confirm. Spawned a dozen of zombies, killed them all, then pulped all their corpses. Then waited for 6 hours - they didn't rezed.

**Comment by dkargin** [comment]
Pulped corpses rise sometimes. I stumble upon this for a month already. I thought it was intended. 
Maybe it happens for the corpses out of the reality bubble.

**Comment by l29ah** [comment]
Those are certainly in.

**Comment by dom4op** [comment]
There might be a zombie necromancer somewhere out of sight

**Comment by Hirmuolio** [comment]
Reviving does not check for pulped status and the pulped and reviving check the corpse damage differently. So I guess they could in some edge case get different result (but I don't think they should be able to).

"pulped" tag is applied by this switch statement:

    switch( damage_level( 4 ) ) {
                case 1:
                    outputstring = pgettext( "damage adjective", "bruised " );
                    break;
                case 2:
                    outputstring = pgettext( "damage adjective", "damaged " );
                    break;
                case 3:
                    outputstring = pgettext( "damage adjective", "mangled " );
                    break;
                default:
                    outputstring = pgettext( "damage adjective", "pulped " );
                    break;
            }

Damage level comes from this:

    int item::damage_level( int max ) const
    {
        if( damage_ == 0 || max <= 0 ) {
            return 0;
        } else if( max_damage() <= 1 ) {
            return damage_ > 0 ? max : damage_;
        } else if( damage_ < 0 ) {
            return -( ( max - 1 ) * ( -damage_ - 1 ) / ( max_damage() - 1 ) + 1 );
        } else {
            return ( max - 1 ) * ( damage_ - 1 ) / ( max_damage() - 1 ) + 1;
        }
    }

Reviving just checks that `damage() >= max_damage()`. If this is true the corpse can't revive.


The pulping action should damage the corpse until `corpse.damage() < corpse.max_damage()` is false. So corpses pulped by pulping action should not revive.

**Comment by dkargin** [comment]
Could max_damage() value be affected by save migrations and become inconsistent with previously saved 'pulped' status?

**Comment by l29ah** [comment]
Necromancers aren't supposed to be able to revive pulped zombies, are they?

**Comment by dom4op** [comment]
> Necromancers aren't supposed to be able to revive pulped zombies, are they?
I'm pretty sure they are supposed to. I could swear I saw this happen on numerous occasions.

**Comment by KurzedMetal** [comment]
I also noticed that sometimes pulped zombies get the yellow highlight (meaning they could rise) in "V" item list view, maybe it's related to what Hirmoulio described too?

I'm also guessing this only happens when the zombie gets pulped by non-player actions? (when you find zombies dead, fight each other, maybe ramming them?)

**Comment by BaumgeistOne** [comment]
In #33303 I wrote down steps to reproduce this bug, at least for the case that a necromancer raises them.

Also, last night I went back into a town a had not visited for a time (so the corpses I had pulped must've decayed somewhat), and a necromancer was able to raise a whole lot of them. Some of them even came back as another type of zombie - for example a boomer (who leaves no corpse when killed).
